docs/general/dql/selects.rst

```diff
@@ -511,274 +502,461 @@ does always return ``NULL`` when comparing ``NULL``.
+    +------------+-----------------------------------------------+
+    SELECT 1 row in set (... sec)
 
-Arrays in CrateDB can be queried for containment using the
-:ref:`sql_dql_any_array` operator.
+The properties of individual child objects can be selected with
+``locations[n]['property']``, where ``n`` is the parent array index and
+``property`` is the child property name, like so::
+
+    cr> select name, information[1]['population'] from locations
+    ... where name = 'Orion Beta';
+    +------------+------------------------------+
+    | name       | information[1]['population'] |
+    +------------+------------------------------+
+    | Orion Beta |                      3600001 |
+    +------------+------------------------------+
+    SELECT 1 row in set (... sec)
 
-It is possible to access fields of :ref:`sql_dql_objects` using subscript
-expressions. If the parent is an object array, you'll get an array of the
-selected field.
+.. NOTE::
 
+    The ``my_column[n]['property']`` syntax can reference the child array index
+    ``n`` of a parent object ``property``, if the column holds :ref:`objects
+    instead of arrays <sql_dql_array_objects>`. This makes the syntax
+    `polymorphic`_ (i.e., behavior is inferred from the data type of the
+    column).
 
-Examples::
+Additionally, you can query the same ``property`` of *every* object within the
+array by omitting the array index, like so::
 
     cr> select name, information['population'] from locations
-    ... where information['population'] is not null
-    ... order by name;
+    ... where information['population'] is not null;
     +-------------------+---------------------------+
     | name              | information['population'] |
     +-------------------+---------------------------+
     | North West Ripple | [12, 42]                  |
     | Outer Eastern Rim | [5673745846]              |
+    | Orion Beta        | [3600001, 1]              |
     +-------------------+---------------------------+
-    SELECT 2 rows in set (... sec)
+    SELECT 3 rows in set (... sec)
 
-::
+.. NOTE::
 
-    cr> select information from locations
-    ... where information['population'] is not null
-    ... order by name;
-    +----------------------------------------------------...-----------------------+
-    | information                                                                  |
-    +----------------------------------------------------...-----------------------+
-    | [{"evolution_level": 4, "population": 12}, {"evolu...": 42, "popul...": 42}] |
-    | [{"evolution_level": 2, "population": 5673745846}]                           |
-    +---------------------------------------------------...------------------------+
+   You can nest arrays and objects however you like, but you can only have one
+   subscript value per expression. The following won't work:
+
+   ``select my_column[1]['property'][2] from my_table;``
```

Comment:
Also see my comment above.
The whole `Object inside arrays` section looks like the repeated version of `Arrays inside objects`. Does this really makes sense like this? Couldn't we fold this together or just highlight important differences and otherwise link to the previous explanations?